app/controllers/bike_share_app.rb

```diff
@@ -1,4 +1,156 @@
+require_relative 'pagination'
+
 class BikeShareApp < Sinatra::Base
 
+  include Pagination
+
+  get '/' do
+    redirect '/stations'
+  end
+
+  get '/stations' do
+    @stations = Station.all
+    erb :"stations/index"
+  end
+
+  get '/stations-dashboard' do
+    erb :"stations/dashboard"
+  end
+
+  get '/stations/new' do
+    @cities = City.all
+    erb :"stations/new"
+  end
+
+  get '/stations/:id' do |id|
+    @station = Station.find(id)
+    erb :"stations/show"
+  end
+
+  post '/stations' do
+    city_name = params["city"]
```

Comment:
if you're just going to use this local variable on the next line, i'd say it's safe to just use params rather than defining a variable here 👍